### PR TITLE
Fix E2E Tests, add CI break on test failure, improve TreeDumpLibrary

### PR DESCRIPTION
**Pull request by asklar**
Improve/refactor e2etest and TreeDumpLibrary in a number of ways:
* most importantly is that we've had test break but not fail the PR or CI - this is because wdio which we use as a test harness, does not set its exit code on success/failure, which is needed for Azure Pipelines to know whether it should fail the build or not.
image
* Because we were blind to test failures, the outputs have drifted away from the original masters. I'm re-baselining them based on what we have today. Component owners should do a pass on the new masters to make sure nothing looks weird (e.g. ImageRTL still shows FlowDirection=LeftToRight). Filed #4589 to track this.
* changes the test app UI to provide links to the master and output files, and to diff the two. I wanted to launch vscode in diff mode for that but since we are a UWP app, the only way we can launch a program like that is to use a protocol handler, which vscode doesn't expose for diffing. Filed: [Provide a protocol handler for diffing two files](https://github.com/microsoft/vscode/issues/95184). For the time being, we'll copy the command line to launch vscode in diff mode for the two files.
* Adds the ability for the tree dump library to output json which will allow us to write tests that validate the output from the javascript side
* Adds logic for optionally output the name of elements. I plan to turn this and json output on soon but wanted to get a baseline PR/CI first.

* e2etest was crashing because of a bug in Switch: #4596 . Disabled that control in the test app for the time being.
* Added logic to the pipeline to capture crash dumps when e2etest app crashes 
* Added checks to make sure we don't try to build/run the e2etest app in scenarios where it isn't really supported because the masters would not match (anything other than Release|x64 at 100% DPI). There are bugs tracking these limitations that we must fix before rtm: #4122 and  #4619 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4585)

### CODE REVIEW

File: packages/E2ETest/windows/ReactUWPTestApp/ReactUWPTestApp.csproj [hunk 1]
```diff
@@ -132,7 +132,12 @@
     <None Include="ReactUWPTestApp_TemporaryKey.pfx" />
   </ItemGroup>
   <ItemGroup>
-    <Content Include="Assets\TreeDump\*.txt" />
+    <Content Include="Assets\TreeDump\ControlStyleRegularBorder.txt" />
+    <Content Include="Assets\TreeDump\ControlStyleRoundBorder.txt" />
+    <Content Include="Assets\TreeDump\ImageRTL.txt" />
+    <Content Include="Assets\TreeDump\ImageWithBorder.txt" />
+    <Content Include="Assets\TreeDump\ImageWithoutBorder-Subsequent.txt" />
+    <Content Include="Assets\TreeDump\ImageWithoutBorder.txt" />
```
Comment (kmelmon) [inline packages/E2ETest/windows/ReactUWPTestApp/ReactUWPTestApp.csproj:140]: Ouch - this won't scale very well.  Why can't we use the wildcard here?

File: packages/E2ETest/windows/ReactUWPTestApp/Assets/TreeDump/ImageRTL.txt [hunk 2]
```diff
@@ -13,16 +13,16 @@
...
-      VerticalAlignment=Stretch
-      Visibility=Visible
-    [Windows.UI.Xaml.DependencyObject]
+    [Microsoft.ReactNative.ViewPanel]
+        Background=[NULL]
+        BorderBrush=#00000000
+        BorderThickness=0,0,0,0
+        Clip=[NULL]
+        CornerRadius=0,0,0,0
+        FlowDirection=LeftToRight
```
Comment (kmelmon) [inline packages/E2ETest/windows/ReactUWPTestApp/Assets/TreeDump/ImageRTL.txt:22]: >FlowDirection=LeftToRight [](start = 8, length = 25)

Saw your earlier comment...  LeftToRight is correct, to match the behavior of the XAML Image element, which explicitly does not mirror images in RTL.  Kai made a change recently to ensure this, which explains why this master hasn't changed.

File: packages/E2ETest/windows/ReactUWPTestApp/Assets/TreeDump/ControlStyleRoundBorder.txt [hunk 3]
```diff
@@ -4,848 +4,921 @@
...
+            [Windows.UI.Xaml.Controls.TextBlock]
                 Clip=[NULL]
-                CornerRadius=2,2,2,2
-                Foreground=#00FFFFFF
-                HorizontalAlignment=Right
-                Margin=0,0,0,0
-                Padding=0,0,0,0
-                RenderSize=0,0
+                FlowDirection=LeftToRight
+                Foreground=#B3E1E1E1
```
Comment (kmelmon) [inline packages/E2ETest/windows/ReactUWPTestApp/Assets/TreeDump/ControlStyleRoundBorder.txt:631]: >Foreground=#B3E1E1E1 [](start = 16, length = 20)

Hmmm I'm not sure why this color changed.

Comment (asklar) [reply packages/E2ETest/windows/ReactUWPTestApp/Assets/TreeDump/ControlStyleRoundBorder.txt:631]: so I'm finding that some of these colors are accent colors. Some of these sizes are changed too and I'm trying to find out why.

One thing that we should note is that the lab machine runs 17763 which is RS5. It's possible changes we depend on might be only available in 19h1+?

File: packages/E2ETest/windows/ReactUWPTestApp/ReactUWPTestApp.csproj (hunk 1 above)
Comment (asklar) [reply packages/E2ETest/windows/ReactUWPTestApp/ReactUWPTestApp.csproj:140]: @kmelmon asterisk is valid in msbuild but VS2019 chokes on it (says item is unparented when trying to do anything with one of the files)

**Pull request merged**